libnavui-tripprogress/src/main/java/com/mapbox/navigation/ui/tripprogress/TripProgressProcessor.kt

```diff
@@ -2,12 +2,53 @@ package com.mapbox.navigation.ui.tripprogress
 
 import com.mapbox.navigation.base.trip.model.RouteProgress
 import com.mapbox.navigation.base.trip.model.RouteProgressState
+import com.mapbox.navigation.utils.internal.ifNonNull
 import java.util.Calendar
 
 internal class TripProgressProcessor {
 
     fun process(action: TripProgressAction): TripProgressResult {
-        return calculateTripProgress(action as TripProgressAction.CalculateTripProgress)
+        return when (action) {
+            is TripProgressAction.CalculateTripDetails -> {
+                calculateTripDetails(action)
+            }
+            is TripProgressAction.CalculateTripProgress -> {
+                calculateTripProgress(action)
+            }
+        }
+    }
+
+    private fun calculateTripDetails(
+        action: TripProgressAction.CalculateTripDetails
+    ): TripProgressResult.TripDetails {
+        val tripDetailsList = mutableListOf<TripProgressResult.RouteLegTripDetail>()
+        ifNonNull(action.route.directionsRoute.legs()) { legs ->
+            legs.forEach { routeLeg ->
+                val eta = routeLeg.duration()?.let { duration ->
+                    Calendar.getInstance().also {
+                        it.add(Calendar.SECOND, duration.toInt())
+                    }.timeInMillis
+                } ?: 0L
+                val legTime = routeLeg.duration() ?: 0.0
+                val legDistance = routeLeg.distance() ?: 0.0
+                tripDetailsList.add(
+                    TripProgressResult.RouteLegTripDetail(
+                        estimatedTimeToArrival = eta,
+                        legTime = legTime,
+                        legDistance = legDistance,
+                    )
+                )
+            }
+        }
+        val totalTime = tripDetailsList.sumOf { it.legTime }
+        val totalDistance = tripDetailsList.sumOf { it.legDistance }
+        val totalEstimatedArrivalTime = tripDetailsList.sumOf { it.estimatedTimeToArrival }
```

Comment:
```suggestion
        val totalTime = action.route.directionsRoute.duration
        val totalDistance = action.route.directionsRoute.distance
        val totalEstimatedArrivalTime = calendar...
```

We don't need to do the calculations ourselves.